### PR TITLE
FEAT: Pagination support for ekplor page

### PR DESCRIPTION
**Pull request by FarhanRafid97**
Closes Pagination in eksplor page #97

## Description

implementasi search public user with pagination using useInfiniteQuery 

## Current Tasks

- [ ] add search input for search public user on explore page
- [ ] add pagination on explore page

## ask
 - [ ]  mas ipan i wanna ask , Is it an issue that the API for the sitemap and the API for the content explore page are distinct cause my lack of experience with sitemaps ? :sweat_smile:

## preview

[d4eb346c-6374-4431-a91c-5ed97863813a.webm](https://github.com/mazipan/tanyaaja/assets/94950541/04c07b55-e1a6-440e-82cc-809d94cd3325)

### CODE REVIEW

**Comment by mazipan** [comment]
Sitemap should have no pagination.
It should list down all the things are available in the site.
Ekplor page on the other hand is the interface for the user, it should paginated.

So it's better to separate the fn.

File: src/lib/utils.ts [hunk 1]
```diff
@@ -213,3 +217,12 @@ export function countQuestion(arr: IResponseGetQuestionPagination[]) {
 
   return totalQuestion
 }
+export function countPublicUser(arr: IResponseGetPublicUserList[]) {
```
Comment (mazipan) [inline src/lib/utils.ts]: How about make it more generic?

Rename the fn name to `calculatePageItemCount`

Comment (FarhanRafid97) [reply src/lib/utils.ts]: done mas

**Review by mazipan — APPROVED** [review]

**Comment by mazipan** [comment]
Could you try to run `pnpm run build` on your local, seems the build is error in Vercel

**Comment by FarhanRafid97** [comment]
di local aman mas 

Tangkapan Layar 2023-10-22 pukul 16 39 20

**Pull request merged**